### ISSUE TITLE
tckresample and tckconvert output not readable by nibabel (dev branch)

### ISSUE THREAD

**Issue by matteofrigo** [MRtrix3/mrtrix3#1925]
Dear all,
I would like to report a but that concerns the nibabel compatibility of the tck files processed with `tckresample` and `tckconvert`.

Let `tracto.tck` be the output of a `tckgen` call, then `tracto-resample.tck` and `tracto-convert.tck` be the two versions of `tracto.tck` that have been processed with `tckresample` and `tckconvert` respectively.
```
tckresample -downsample 4 tracto.tck tracto-resample.tck
tckconvert tracto.tck tracto-convert.tck
```

In a python shell, I execute the following code:
```
import nibabel as nib
a = nib.streamlines.load('tracto.tck')
b = nib.streamlines.load('tracto-resample.tck')
c = nib.streamlines.load('tracto-convert.tck')
```
Variable `a` loads correctly, while both `b` and `c` produce the same error:
```
In [4]: b = nib.streamlines.load('tracto-convert.tck')                                                                                      
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-b2ccf9ef05a7> in <module>
----> 1 b = nib.streamlines.load('tracto-convert.tck')

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/__init__.py in load(fileobj, lazy_load)
     94         raise ValueError("Unknown format for 'fileobj': {}".format(fileobj))
     95 
---> 96     return tractogram_file.load(fileobj, lazy_load=lazy_load)
     97 
     98 

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/tck.py in load(cls, fileobj, lazy_load)
    141         have referred to a corner.
    142         """
--> 143         hdr = cls._read_header(fileobj)
    144 
    145         if lazy_load:

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/tck.py in _read_header(fileobj)
    323 
    324         # Build header dictionary from the buffer.
--> 325         hdr = dict(item.split(': ') for item in buf.rstrip().split('\n')[:-1])
    326         hdr[Field.MAGIC_NUMBER] = magic_number
    327 

ValueError: dictionary update sequence element #3 has length 1; 2 is required
```

The header of the three files is the following (`head -n 33 tracto*.tck`)
_tracto.tck_
```
mrtrix tracks    
act: /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz
backtrack: 1
command_history: tckgen -algorithm iFOD2 -select 2K -act /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz -downsample 4 -minlength 40 -maxlength 400 -seed_gmwmi /user/mfrigo/home/projects/fit/data/phantom/tractography/gmwmi.nii.gz -backtrack -crop_at_gmwmi -force fodf.mif streamlines.tck  (version=3.0_RC4-391-gb154f617)
crop_at_gmwmi: 1
downsample_factor: 4
fod_power: 0.25
init_threshold: 0.0500000007
lmax: 8
max_angle: 45
max_dist: 400
max_num_seeds: 2000000
max_num_tracks: 2000
max_seed_attempts: 1000
max_trials: 1000
method: iFOD2
min_dist: 40
mrtrix_version: 3.0_RC4-391-gb154f617
rk4: 0
samples_per_step: 4
sh_precomputed: 1
source: fodf.mif
step_size: 0.625
stop_on_all_include: 0
threshold: 0.0500000007
timestamp: 1580289707.4327952862
unidirectional: 0
roi: seed gmwmi.nii.gz
datatype: Float32LE
file: . 1008
count: 2000
total_count: 2395
END
```
_tracto-resample.tck_
```
mrtrix tracks    
act: /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz
backtrack: 1
command_history: tckgen -algorithm iFOD2 -select 2K -act /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz -downsample 4 -minlength 40 -maxlength 400 -seed_gmwmi /user/mfrigo/home/projects/fit/data/phantom/tractography/gmwmi.nii.gz -backtrack -crop_at_gmwmi -force fodf.mif streamlines.tck  (version=3.0_RC4-391-gb154f617)
tckresample -downsample 4 streamlines.tck streamlines-downsample.tck  (version=3.0_RC4-391-gb154f617)
crop_at_gmwmi: 1
downsample_factor: 4
fod_power: 0.25
init_threshold: 0.0500000007
lmax: 8
max_angle: 45
max_dist: 400
max_num_seeds: 2000000
max_num_tracks: 2000
max_seed_attempts: 1000
max_trials: 1000
method: iFOD2
min_dist: 40
mrtrix_version: 3.0_RC4-391-gb154f617
rk4: 0
samples_per_step: 4
sh_precomputed: 1
source: fodf.mif
step_size: 0.625
stop_on_all_include: 0
threshold: 0.0500000007
timestamp: 1580289769.6081135273
unidirectional: 0
prior_roi: seed gmwmi.nii.gz
datatype: Float32LE
file: . 1116
count: 2000
total_count: 2000
```
_tracto-convert.tck_
```
mrtrix tracks    
act: /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz
backtrack: 1
command_history: tckgen -algorithm iFOD2 -select 2K -act /user/mfrigo/home/projects/fit/data/phantom/tractography/fivett.nii.gz -downsample 4 -minlength 40 -maxlength 400 -seed_gmwmi /user/mfrigo/home/projects/fit/data/phantom/tractography/gmwmi.nii.gz -backtrack -crop_at_gmwmi -force fodf.mif streamlines.tck  (version=3.0_RC4-391-gb154f617)
tckconvert streamlines.tck streamlines-tckconvert.tck  (version=3.0_RC4-391-gb154f617)
crop_at_gmwmi: 1
downsample_factor: 4
fod_power: 0.25
init_threshold: 0.0500000007
lmax: 8
max_angle: 45
max_dist: 400
max_num_seeds: 2000000
max_num_tracks: 2000
max_seed_attempts: 1000
max_trials: 1000
method: iFOD2
min_dist: 40
mrtrix_version: 3.0_RC4-391-gb154f617
rk4: 0
samples_per_step: 4
sh_precomputed: 1
source: fodf.mif
step_size: 0.625
stop_on_all_include: 0
threshold: 0.0500000007
timestamp: 1580289859.7659745216
unidirectional: 0
prior_roi: seed gmwmi.nii.gz
datatype: Float32LE
file: . 1100
count: 2000
total_count: 2000
```
I noticed that there is a double line for the `command_history` field, so I tried to manually delete the second one with vim. It didn't solve the problem, but now the python error is different:
```
In [2]: a = nib.streamlines.load('tracto-convert-nodoubleline.tck')                                                                                       
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-ed3250afce34> in <module>
----> 1 a = nib.streamlines.load('tracto-convert-nodoubleline.tck')

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/__init__.py in load(fileobj, lazy_load)
     94         raise ValueError("Unknown format for 'fileobj': {}".format(fileobj))
     95 
---> 96     return tractogram_file.load(fileobj, lazy_load=lazy_load)
     97 
     98 

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/tck.py in load(cls, fileobj, lazy_load)
    152         else:
    153             tck_reader = cls._read(fileobj, hdr)
--> 154             streamlines = ArraySequence(tck_reader)
    155             tractogram = Tractogram(streamlines)
    156 

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/array_sequence.py in __init__(self, iterable, buffer_size)
    101             return
    102 
--> 103         self.extend(iterable)
    104 
    105     @property

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/array_sequence.py in extend(self, elements)
    248                                  self._build_cache)
    249 
--> 250         for e in elements:
    251             self.append(e, cache_build=True)
    252 

~/miniconda3/lib/python3.7/site-packages/nibabel/streamlines/tck.py in _read(cls, fileobj, header, buffer_size)
    408                     buff = buff[:n_read]
    409 
--> 410                 raw_values = np.frombuffer(buff, dtype=dtype)
    411 
    412                 # Convert raw_values into a list of little-endian triples (for x,y,z coord)

ValueError: buffer size must be a multiple of element size
```

To me it looks like when the tck goes through `tckconvert` or `tckresample` the header is compromised in some critical way.

Mrtrix version: 3.0_RC4-391-gb154f617
Nibabel version: 2.4.0 on py3 (conda installation)
Platform:
```
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	Fedora
Description:	Fedora release 31 (Thirty One)
Release:	31
Codename:	ThirtyOne
```

_Note: The bug does not show if I run everything with the mrtrix compiled from the master branch_

Thank you in advanced for your help.
Regards,
Matteo

**Comment by Lestropie** [comment]
My bad :raising_hand_man: 

About to go to bed, but pretty sure #1926 is all that's needed?

**Comment by matteofrigo** [comment]
It will most probably do the job. Thanks!

**Comment by matteofrigo** [comment]
Solved by #1926 .
